src/streamlink/plugins/afreeca.py

```diff
@@ -224,10 +215,12 @@ def _get_streams(self):
         self.author = channel.get("BJNICK")
         self.title = channel.get("TITLE")
 
-        for qkey in self.QUALITYS:
-            hls_stream = self._get_hls_stream(broadcast, username, qkey, rmd)
+        for qkey in channel.get("VIEWPRESET"):
+            if qkey.get("name") == "auto":
+                continue
+            hls_stream = self._get_hls_stream(broadcast, username, qkey.get("name"), rmd)
             if hls_stream:
-                yield qkey, hls_stream
+                yield qkey.get("label"), hls_stream
```

Comment:
The `.get()` calls are unnecessary (and could unintentionally return `None` if the key were missing - not possible due to the validation schema though). Just use regular `__getitem__` subscript syntax: `obj[key]`

Reply:
So it's okay to modify all the other .get's as long as they work?

Reply:
The other ones are intentional because the data is maked as optional in the validation schema. The validation schema handles two different kinds of API responses: error responses and valid data responses. And optional metadata which you've added in February.